internal/actions/penalize.go

```diff
@@ -0,0 +1,31 @@
+package actions
+
+import (
+	"github.com/NodeFactoryIo/vedran/internal/models"
+	"github.com/NodeFactoryIo/vedran/internal/repositories"
+	"github.com/NodeFactoryIo/vedran/internal/schedule/penalize"
+	log "github.com/sirupsen/logrus"
+)
+
+const InitialPenalizeIntervalInMins = 1
+
+// PenalizeNode removes provided node from active nodes, sets initial cooldown of 1 minute and schedules check for
+// penalized node by invoking penalize.ScheduleCheckForPenalizedNode
+func (a *actions) PenalizeNode(node models.Node, repositories repositories.Repos) {
+	// remove node from active
+	err := repositories.NodeRepo.RemoveNodeFromActive(node)
+	if err != nil {
+		log.Errorf("Failed penalizing node because of: %v", err)
+		return
+	}
+
+	// set new cooldown
+	node.Cooldown = InitialPenalizeIntervalInMins
+	err = repositories.NodeRepo.Save(&node)
+	if err != nil {
+		log.Errorf("Failed penalizing node because of: %v", err)
```

Comment:
if it fails, it's gonna print it's penalized and schedulecheck?

Reply:
I will add `return` after log, and expand both logs to display node ID

`log.Errorf("Failed penalizing node %s because of: %v", node.ID, err)`